galter_subjects_utils/updater.py

```diff
@@ -152,49 +185,40 @@ def get_records_to_update(ops_data, data_cls):
     """Return data-layer records to update."""
 
     def get_targeted_ids(ops_data):
-        relevant_ops = (
-            filter_ops_by_type(ops_data, "replace") +
-            filter_ops_by_type(ops_data, "remove") +
-            filter_ops_by_type(ops_data, "rename")
-        )
-        return [op["id"] for op in relevant_ops]
-
-    def at_least_1_subject_targeted(ids):
-        """Where clause to get records with at least 1 relevant subject id.
+        return [
+            op["id"] for op in ops_data
+            if op.get("type") in ["replace", "remove", "rename"]
+        ]
 
-        Only call this for non-empty `ids`, otherwise the resulting where
-        clause is all-permissive.
+    def has_at_least_1_subject_targeted(record_data_db, ids):
+        """Return True if `record_data_db` has at least 1 subject in `ids`.
 
-        WARNING: This is very ugly and I don't know how to make it better.
-                 We use RAW SQL to check for presence of id (@>). The syntax
-                 is not great but it gives us what we want.
+        Because of cases where there are 100K+ subjects, we can't construct
+        queries filtering at the DB level. We do the filtering in memory,
+        on batches of records. We keep loading simple at the cost of
+        performance (but this is fine since this operation is in the
+        background anyway).
```

Comment:
Thanks for the explanation. It's a lot easier to look at this way even if it is less performant.